tests/CoordinateServiceTest.cs

```diff
@@ -0,0 +1,197 @@
+﻿using BDMS.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace BDMS;
+
+[TestClass]
+public class CoordinateServiceTest
+{
+    private const int LV95BoreholeWithAllCoordinatesSetId = 2_000_000;
+    private const int LV03BoreholeWithMissingDestCoordinatesId = 2_000_001;
+    private const int LV03BoreholeWithMissingSourceCoordinatesId = 2_000_002;
+
+    private BdmsContext context;
+    private CoordinateService service;
+    private Mock<IHttpClientFactory> httpClientFactoryMock;
+    private Mock<ILogger<CoordinateService>> loggerMock;
+    private List<int> boreholeIds = new List<int> { LV95BoreholeWithAllCoordinatesSetId, LV03BoreholeWithMissingDestCoordinatesId, LV03BoreholeWithMissingSourceCoordinatesId };
+
+    [TestInitialize]
+    public void TestInitialize()
+    {
+        context = ContextFactory.CreateContext();
+        httpClientFactoryMock = new Mock<IHttpClientFactory>(MockBehavior.Strict);
+        loggerMock = new Mock<ILogger<CoordinateService>>();
+
+        service = new CoordinateService(loggerMock.Object, httpClientFactoryMock.Object);
+    }
+
+    [TestCleanup]
+    public async Task TestCleanup()
+    {
+        // Remove created Boreholes
+        context.Boreholes.Where(b => boreholeIds.Contains(b.Id)).ToList().ForEach(borehole =>
+            context.Boreholes.Remove(borehole));
+        context.SaveChanges();
+
+        await context.DisposeAsync();
+        httpClientFactoryMock.Verify();
+        loggerMock.Verify();
+    }
+
+    [TestMethod]
+    public void GetDecimalPlaces()
+    {
+        Assert.AreEqual(3, CoordinateService.GetDecimalPlaces(1234567.123));
+        Assert.AreEqual(6, CoordinateService.GetDecimalPlaces(100000.123456));
+        Assert.AreEqual(2, CoordinateService.GetDecimalPlaces(10000000.01));
+        Assert.AreEqual(0, CoordinateService.GetDecimalPlaces(1000000));
+    }
+
+    [TestMethod]
+    public async Task MigrateCoordinatesOfLV95BoreholeWithAllCoordinatesSet()
+    {
+        httpClientFactoryMock.Setup(cf => cf.CreateClient(It.IsAny<string>())).Returns(new HttpClient()).Verifiable();
+
+        var borehole = CreateLV95BoreholeWithAllCoordinatesSet();
+
+        await service.MigrateCoordinatesOfBorehole(borehole, false);
+        context.SaveChanges();
+
+        var result = context.Boreholes.FirstOrDefault(b => b.Id == LV95BoreholeWithAllCoordinatesSetId);
+        Assert.AreEqual(ReferenceSystem.LV95, result.OriginalReferenceSystem);
+        Assert.AreEqual(2626103.1988343936, result.LocationX);
+        Assert.AreEqual(1125366.3469565178, result.LocationY);
+        Assert.AreEqual(626103.56923180178, result.LocationXLV03);
+        Assert.AreEqual(125366.57802526229, result.LocationYLV03);
+    }
+
+    [TestMethod]
+    public async Task NotMigrateCoordinatesOfLV95BoreholeWithAllCoordinatesSet()
```

Comment:
PP: Der Methodenname klingt ein wenig holprig. Alternativvorschlag:
```suggestion
    public async Task DoesNotMigrateCoordinatesOfLV95BoreholeWithAllCoordinatesSet()
```